src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -508,6 +512,32 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 		);
 	}
 
+	async removeCompetency(competency: this['badgeClassForm']['controls']['competencies']['controls'][0]) {
+		const value = competency.value;
+
+		if (
+			(value.name || '').trim().length > 0 ||
+			(value.description || '').trim().length > 0 ||
+			(value.escoID || '').trim().length > 0 ||
+			(value.category || '').trim().length > 0
+		) {
+			if (
+				!(await this.dialogService.confirmDialog.openTrueFalseDialog({
+					dialogTitle: 'Remove Competency?',
+					dialogBody: 'Are you sure you want to remove this competency? This action cannot be undone.',
+					resolveButtonLabel: 'Remove Competency',
+					rejectButtonLabel: 'Cancel',
+				}))
+			) {
+				return;
+			}
+		}
+
+		this.badgeClassForm.controls.competencies.removeAt(
```

Comment:
It seems that this should be inside the body of the previous if, because otherwise this will try to remove an empty competence

Reply:
Ah I see. But then the if structure is a bit weird; like, why are there two ifs?

Reply:
Tbh i basically copied the ```removeAlignment``` function, i am not sure why this structure was chosen. I would suggest to leave it as is for now, but to combine the functions once we are certain that the badge creation process is finished. What do you think, should i add a _TODO_ and ticket for that?

Reply:
If it's the same then I think it is alright